Library/Homebrew/formula.rb

```diff
@@ -261,6 +261,15 @@ def installed?
     (dir = installed_prefix).directory? && dir.children.length > 0
   end
 
+  # If this {Formula} has been modified by the user.  Relies on git.
+  def adulterated?
+    repo = core_formula? ? HOMEBREW_LIBRARY : path_of_repository
+    repo.cd do
+      return true unless `git status --porcelain #{path}`.empty?
```

Comment:
You need to check whether the file is under git control. Also please use `Utils.popen_read("git", "status", ...)` instead of backticks to avoid shell injection.